click/_compat.py

```diff
@@ -512,7 +519,19 @@ def open_stream(filename, mode='r', encoding=None, errors='strict',
     else:
         f = os.fdopen(fd, mode)
 
-    return _AtomicFile(f, tmp_filename, os.path.realpath(filename)), True
+    real_filename = os.path.realpath(filename)
+
+    # Get permissions to set on the target file. If the target file already
+    # exists, retain its current permissions. If the target file will be
+    # created, respect the permissions from the current umask.
+    permissions = None
+    if not WIN:
+        try:
+            permissions = stat.S_IMODE(os.stat(real_filename).st_mode)
+        except OSError:
```

Comment:
```suggestion
        except (OSError, IOError):
```

Handle `IOError` for py2 compatibility.

Reply:
baded409cfef144e3acd0e8d8f3c7c7e197723e7